pkg/kv/kvclient/rangefeed/rangefeed_external_test.go

```diff
@@ -876,3 +876,59 @@ func TestRangefeedWithLabelsOption(t *testing.T) {
 	defer allLabelsCorrect.Unlock()
 	require.True(t, allLabelsCorrect.correct)
 }
+
+// TestRangeFeedStartTimeExclusive tests that the start timestamp of the
+// rangefeed is in fact exclusive, as specified.
+func TestRangeFeedStartTimeExclusive(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	ctx := context.Background()
+	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{})
+	defer tc.Stopper().Stop(ctx)
+
+	srv0 := tc.Server(0)
+	db := srv0.DB()
+	scratchKey := tc.ScratchRange(t)
+	scratchKey = scratchKey[:len(scratchKey):len(scratchKey)]
+	mkKey := func(k string) roachpb.Key {
+		return encoding.EncodeStringAscending(scratchKey, k)
+	}
+	span := roachpb.Span{Key: scratchKey, EndKey: scratchKey.PrefixEnd()}
+
+	// Write three versions of "foo". Get the timestamp of the second version.
+	require.NoError(t, db.Put(ctx, mkKey("foo"), 1))
+	require.NoError(t, db.Put(ctx, mkKey("foo"), 2))
+	kv, err := db.Get(ctx, mkKey("foo"))
+	require.NoError(t, err)
+	ts2 := kv.Value.Timestamp
+	require.NoError(t, db.Put(ctx, mkKey("foo"), 3))
+
+	// Enable rangefeeds, otherwise the thing will retry until they are enabled.
```

Comment:
Interesting... I wonder if we should not do that.  System ranges against which this library is used already are exempt; but user ranges... we are probably better off returning an error... Just food for thought.

Reply:
Opened #82512